### PR TITLE
add support for path type

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

File: redisgraph/query_result.py [hunk 1]
```diff
@@ -124,6 +126,12 @@ def parse_edge(self, cell):
         properties = self.parse_entity_properties(cell[4])
         return Edge(src_node_id, relation, dest_node_id, edge_id=edge_id, properties=properties)
 
+
```
Comment (swilly22) [inline redisgraph/query_result.py]: Remove empty line

File: redisgraph/query_result.py [hunk 2]
```diff
@@ -124,6 +126,12 @@ def parse_edge(self, cell):
...
+    def parse_path(self, cell):
+        nodes = self.parse_scalar(cell[0])
+        relationships = self.parse_scalar(cell[1])
+        return Path(nodes, relationships)
```
Comment (swilly22) [inline redisgraph/query_result.py]: Assert both nodes and relationships types, consider performing the assertion in Path's constructor

File: redisgraph/path.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+
+class Path(object):
+
+    def __init__(self, nodes, relationships):
+        self.nodes = nodes
+        self.relationships = relationships
+
```
Comment (swilly22) [inline redisgraph/path.py]: Please add `def length`

File: redisgraph/query_result.py (hunk 1 above)
Comment (DvirDukhan) [reply redisgraph/query_result.py]: done

File: redisgraph/path.py (hunk 3 above)
Comment (DvirDukhan) [reply redisgraph/path.py]: done

File: test.py [hunk 4]
```diff
@@ -65,5 +65,35 @@ def test_array_functions(self):
...
+        edge01 = Edge(node0, "R1", node1, edge_id=0, properties={'value': 1})
+        edge12 = Edge(node1, "R1", node2, edge_id=1, properties={'value': 2})
+
+        redis_graph.add_node(node0)
+        redis_graph.add_node(node1)
+        redis_graph.add_node(node2)
+        redis_graph.add_edge(edge01)
+        redis_graph.add_edge(edge12)
+
+        redis_graph.commit()
```
Comment (swilly22) [inline test.py]: .flush()

File: redisgraph/path.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        self.append_type = Node
+
+    @classmethod
+    def new_empty_path(cls):
+        return cls([], [])
+
+    @classmethod
+    def new_path(cls, nodes, edges):
+        return cls(nodes, edges)
+
```
Comment (swilly22) [inline redisgraph/path.py]: Remove this function, use constructor.

File: redisgraph/path.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    def edge_count(self):
+        return len(self.edges)
+    
+    def nodes_count(self):
+        return len(self.nodes)
+
+    def add_node(self, node):
+        assert(isinstance(node, self.append_type))
+        self.nodes.append(node)
+        self.append_type = Edge
```
Comment (swilly22) [inline redisgraph/path.py]: `self.append_type = type(Edge)` seems more aligned with the naming.
but then  `assert(isinstance(node, self.append_type))` should be updated to
` assert(type(node) == self.append_type)`

File: redisgraph/path.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        return self
+
+    def add_edge(self, edge):
+        assert(isinstance(edge, self.append_type))
```
Comment (swilly22) [inline redisgraph/path.py]: See above comment.

File: README.md [hunk 8]
```diff
@@ -47,6 +47,16 @@ for record in result.result_set:
...
+	
+query = """MATCH p = (:person)-[:visited {purpose:"pleasure"}]->(:country) RETURN p"""
+
+result = redis_graph.query(query)
+
+# Iterate through resultset
+for record in result.result_set:
+    path = record[0]
+    # See path.py for more path API.
+    print(path.edge_count(), path.nodes_count())
```
Comment (swilly22) [inline README.md]: The example should print the path, don't refer users to source code within the readme.

File: README.md [hunk 9]
```diff
@@ -47,6 +47,16 @@ for record in result.result_set:
...
 	country_name = record[3]
+	
+query = """MATCH p = (:person)-[:visited {purpose:"pleasure"}]->(:country) RETURN p"""
+
+result = redis_graph.query(query)
+
+# Iterate through resultset
+for record in result.result_set:
+    path = record[0]
+    # See path.py for more path API.
```
Comment (swilly22) [inline README.md]: Please remove this comment.

**Pull request merged**